fire_forecast/fire_model/fire_model.py

```diff
@@ -0,0 +1,136 @@
+"""
+Supply a fire model that creates fire time series data. 
+
+The idea is that the forecasting algorithms can be trained on the synthetic.
+"""
+
+import numpy as np
+import xarray as xr
+
+def create_ts_temperature(t, rng=None, seed=0):
+    """
+    Create a temperature time series.
+
+    The time series is created for hourly time steps. The temperature includes a diurnal
+    cycle and a random trend with additional noise.
+
+    Parameters
+    ----------
+    t : int
+        The length of the time series
+    rng : np.random.Generator, optional
+        The random number generator used by the function. If not specified, a
+        new generator is created.
+    seed : int, optional
+        The seed for the random number generator. Only used if rng is not
+        specified.
+
+    Returns
+    -------
+    temperature : ndarray
+        The time series of the temperature
+    """
+    if rng is None:
+        rng = np.random.default_rng(seed=seed)
+    # Create the diurnal cycle
+    t_diurnal_cycle = 24
+    diurnal_cycle = 5 * np.sin(np.arange(t) * 2 * np.pi / t_diurnal_cycle)
+    # Create the random trend
+    offset = rng.random() * 10
+    trend = np.linspace(offset, offset + rng.random() * 10, t)
+    # Create the noise
+    noise = rng.normal(0, 2, t)
+    # Combine the components
+    temperature = diurnal_cycle + trend + noise
+    return temperature
+
+def create_ts_humidity(t, rng=None, seed=0):
+    """
+    Create a humidity time series.
+
+    The time series is created for hourly time steps. The humidity includes a diurnal
+    cycle and a random trend with additional noise.
+
+    Parameters
+    ----------
+    t : int
+        The length of the time series
+    rng : np.random.Generator, optional
+        The random number generator used by the function. If not specified, a
+        new generator is created.
+    seed : int, optional
+        The seed for the random number generator. Only used if rng is not
+        specified.
+
+    Returns
+    -------
+    humidity : ndarray
+        The time series of the humidity
+    """
+    if rng is None:
+        rng = np.random.default_rng(seed=seed)
+    # Create the diurnal cycle
+    t_diurnal_cycle = 24
+    diurnal_cycle = 5 * np.sin(np.arange(t) * 2 * np.pi / t_diurnal_cycle)
+    # Create the random trend
+    offset = rng.random() * 30
+    trend = np.linspace(offset, offset + rng.random() * 10, t)
+    # Create the noise
+    noise = rng.normal(0, 2, t)
+    # Combine the components
+    humidity = diurnal_cycle + trend + noise + 10
```

Comment:
Wie oben vllt. die Zahlen als variablen, damit es automatisch verständlich ist. Kann vllt auch in eine eigene Funktion weil hier der Code doppelt ist im Vergleich zu `create_ts_humidity`